plat/common/arm/pauth.c

```diff
@@ -72,27 +28,40 @@ static inline bool pauth_supported(void)
 	apa = (reg >> ID_AA64ISAR1_EL1_APA_SHIFT) & ID_AA64ISAR1_EL1_APA_MASK;
 	api = (reg >> ID_AA64ISAR1_EL1_API_SHIFT) & ID_AA64ISAR1_EL1_API_MASK;
 	if (apa || api)
-		return true;
+		return __true;
 
 	reg = SYSREG_READ(ID_AA64ISAR2_EL1);
 	apa3 = (reg >> ID_AA64ISAR2_EL1_APA3_SHIFT) & ID_AA64ISAR2_EL1_APA3_MASK;
 	if (apa3)
-		return true;
+		return __true;
 
-	return false;
+	return __false;
 }
 
 int __no_pauth ukplat_pauth_init(void)
 {
 	__u64 key_hi, key_lo;
 	__u64 reg;
+	int rc;
 
-	if (unlikely(!pauth_supported()))
+	if (unlikely(!pauth_supported())) {
+		uk_pr_err("The CPU does not implement FEAT_PAUTH\n");
 		return -ENOTSUP;
+	}
 
 	/* Program instruction Key A */
-	ukplat_pauth_gen_key(&key_hi, &key_lo);
+	rc = uk_random_fill_buffer(&key_hi, sizeof(key_hi));
+	if (unlikely(rc)) {
+		uk_pr_err("Could not generate APIAKey (%d)\n", rc);
```

Comment:
Also here and the message at line 62, should we say `"Failed to ..."`? Again probably preference and I am not strongly object `"Could not..."`.

Reply:
See https://github.com/unikraft/unikraft/pull/1496#discussion_r1858175667